src/frontend/app/store/reducers/pagination-reducer/pagination-reducer-clear-pagination-type.ts

```diff
@@ -9,29 +19,55 @@ import {
 } from '../../helpers/entity-factory';
 import { PaginationState } from '../../types/pagination.types';
 
-export function paginationClearType(state: PaginationState, entityKey, defaultPaginationEntityState) {
-  if (state[entityKey]) {
-    const clearState = { ...state };
-    const entityState = clearState[entityKey];
-    const newObject = {};
-    Object.keys(entityState).forEach(key => {
-      newObject[key] = defaultPaginationEntityState;
-    });
-    clearState[entityKey] = newObject;
-    return clearState;
-  }
-  return state;
+export function paginationClearAllTypes(state: PaginationState, entityKeys: string[], defaultPaginationEntityState) {
+  return entityKeys.reduce((prevState, entityKey) => {
+    if (prevState[entityKey]) {
+      const entityState = state[entityKey];
+      const clearedEntity = Object.keys(entityState).reduce((prevEntityState, key) => {
+        return {
+          ...prevEntityState,
+          [key]: defaultPaginationEntityState
```

Comment:
Regarding the previous code.. when we come back to this in the future we should just remove the entry instead of resetting.